api/test/e2enew/service/service_test.go

```diff
@@ -0,0 +1,455 @@
+		t := ginkgo.GinkgoT()
+		var createServiceBody map[string]interface{} = map[string]interface{}{
+			"name": "testservice",
+			"upstream": map[string]interface{}{
+				"type": "roundrobin",
+				"nodes": []map[string]interface{}{
+					{
+						"host":   base.UpstreamIp,
+						"port":   1980,
+						"weight": 1,
+					},
+					{
+						"host":   base.UpstreamIp,
+						"port":   1981,
+						"weight": 2,
+					},
+					{
+						"host":   base.UpstreamIp,
+						"port":   1982,
+						"weight": 3,
+					},
+				},
+			},
+		}
+		_createServiceBody, err := json.Marshal(createServiceBody)
+		assert.Nil(t, err)
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodPut,
+			Path:         "/apisix/admin/services/s1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			Body:         string(_createServiceBody),
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   []string{"\"id\":\"s1\"", "\"name\":\"testservice\""},
+		})
+	})
+	ginkgo.It("get the service s1", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:     base.ManagerApiExpect(),
+			Method:     http.MethodGet,
+			Path:       "/apisix/admin/services/s1",
+			Headers:    map[string]string{"Authorization": base.GetToken()},
+			ExpectCode: http.StatusOK,
+			ExpectBody: "\"name\":\"testservice\",\"upstream\":{\"nodes\":[{\"host\":\"" + base.UpstreamIp + "\",\"port\":1980,\"weight\":1},{\"host\":\"" + base.UpstreamIp + "\",\"port\":1981,\"weight\":2},{\"host\":\"" + base.UpstreamIp + "\",\"port\":1982,\"weight\":3}],\"type\":\"roundrobin\"}",
+		})
+	})
+	ginkgo.It("create route using the service just created", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/server_port",
+				"service_id": "s1"
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			Sleep:        base.SleepTime,
+		})
+	})
+	ginkgo.It("batch test /server_port api", func() {
+		t := ginkgo.GinkgoT()
+		time.Sleep(time.Duration(500) * time.Millisecond)
+		res := base.BatchTestServerPort(18)
+		assert.True(t, res["1980"] == 3)
+		assert.True(t, res["1981"] == 6)
+		assert.True(t, res["1982"] == 9)
+	})
+	ginkgo.It("delete route", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		})
+	})
+	ginkgo.It("delete service", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/services/s1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
```

Comment:
Need to sleep here, because it may check whether there is a bound route.

Reply:
done.